WordPress/gutenberg
# Add new private `upload-media` package

## What?


This is a new package for client-side media processing, see #61447 for full context.

It implements a queue-like system for processing media items prior to uploading them to the server. This will be used for things like resizing, cropping, compressing, thumbnail generation, poster generation, subtitle generation, etc.

See also https://github.com/swissspidy/media-experiments/blob/7b91599369f1d3ae412e3e18a4afe2f62257b4dc/docs/technical-overview.md.

This logic has been tested extensively as part of https://github.com/swissspidy/media-experiments, but for this initial PR several features have been removed. They will be reintroduced later in separate PRs to make reviews easier.

The package is private and shall not be published until later on when all of the subsequent PRs are merged.

## Why?


Required for a new era of upload handling in Gutenberg

## How?


Adds a new package to facilitate integration in the block-editor package

## Testing Instructions





CI should be green.

Enable the client-side media processing experiment and upload media to the editor. Everything should still work as usual.

### Testing Instructions for Keyboard


N/A

## Screenshots or screencast 

N/A
